app/controllers/batch_controller.rb

```diff
@@ -32,5 +36,12 @@ def initialize_fields(file)
     file.initialize_fields
   end
 
+  def uploading_on_behalf_of? batch
+    return false if batch.generic_files.count == 0  
+    on_behalf_of = batch.generic_files[0].on_behalf_of
+    return false if on_behalf_of == nil or on_behalf_of.empty?
+    current_user.email != batch.generic_files[0].on_behalf_of
```

Comment:
You never want to hard_code email, because some projects use a different user identifier (such as Scholarsphere). Use `current_user.user_key` instead.